crux_core/src/bridge/mod.rs

```diff
@@ -184,19 +193,16 @@ where
     /// # Errors
     ///
     /// Returns an error if the event could not be deserialized.
-    pub fn process_event<'de, D, S>(&self, event: D, requests_out: S) -> Result<(), BridgeError>
+    pub fn process_event<'a>(
+        &self,
+        event: &'a [u8],
+        requests_out: &mut Vec<u8>,
+    ) -> Result<(), BridgeError<T>>
```

Comment:
Is the return via mutable argument essentially the only difference between the two bridge implementations? 

If so, can we also rename `BridgeWithSerializer` to something more descriptive? 

More broadly, I'm quite tempted to say let's just make this interface the default - it's a small change in the FFI and allows more flexibility in the memory ownerhship.

@StuartHarris what do you think, can you think of any downsides?